tests/libsecret/secret.go

```diff
@@ -35,6 +40,27 @@ func New(name string, data map[string][]byte) *kubev1.Secret {
 				secretLabel: name,
 			},
 		},
-		Data: data,
+		Data:       data.Data(),
+		StringData: data.StringData(),
 	}
 }
+
+type DataBytes map[string][]byte
+
+func (s DataBytes) Data() map[string][]byte {
+	return s
+}
+
+func (DataBytes) StringData() map[string]string {
+	return nil
+}
```

Comment:
IMO its not straight forward to understand, each type of map has to implement both `Data` and `StringData` but it wont make sense for `DataBytes` to implement `StringData` (and the other way around).

How about applying builder pattern and expose `WithData` and `WithStringData` options instead?
For example:
```
libsecret.New("mysecret", libsecret.WithStringData(map[string]string{"userdata": userData})
```

Reply:
This is a pure OO design, having an interface and concrete implementations to implements the needed.

I think it is easier to consume as it is now. Using a build pattern can work, but it will be odd to use it for two options only and it also is much noisier than this proposal.

Reply:
IMO pure or not, the implementation is confusing. 

Regrading ease of use, I think its neglectable in this case (for both solutions) eventually you pass a func or a map.
Note that vast majority of usages pass map[string]string, they can use 'StringData'  directly.
The other few places that pass bytes can convert to string.